### PR TITLE
BUG: Cleanup timedelta offset

### LINKED ISSUE #21762: pd.Timedelta() cannot process offset alias T while pd.infer_freq() return this format.
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
series = pd.Series([1, 2, 3], pd.date_range('201701010000', '201701010030', freq='15min'))
print(pd.infer_freq(series.index))
# Error
pd.Timedelta(pd.infer_freq(series.index))
# While this works
pd.Timedelta(series.index.freq)
```
#### Problem description

pd.Timedelta constructor works with e.g. '15min' but not with '15T',
while pd.infer_freq() returns '15T' and pd.Series.index.freq returns '15min'.
This leads to errors.

#### Expected Output

Same behaviour of pd.Timedelta() for 'min' and 'T' offset alias or deprecation of 'T' as an offset alias.

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.4.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 142 Stepping 9, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None
pandas: 0.23.1
pytest: 3.6.2
pip: 10.0.1
setuptools: 39.2.0
Cython: None
numpy: 1.14.5
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 6.4.0
sphinx: None
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.2.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: None
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] closes #21762
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Fixed to use same unit parsing logic in `Timedelta` and `to_timedelta`. Needs to update after #23264 and #23259.

### CODE REVIEW

File: pandas/_libs/tslibs/timedeltas.pxd [hunk 1]
```diff
@@ -6,3 +6,6 @@ from numpy cimport int64_t
 cdef int64_t cast_from_unit(object ts, object unit) except? -1
 cpdef int64_t delta_to_nanoseconds(delta) except? -1
 cpdef convert_to_timedelta64(object ts, object unit)
+
+# Exposed for Timedelta, not intended for outside use.
```
Comment (jbrockmendel) [inline pandas/_libs/tslibs/timedeltas.pxd]: Is this cimported by another cython module?  If not, doesn’t need to be in the pxd file

File: pandas/tests/scalar/timedelta/test_timedelta.py [hunk 2]
```diff
@@ -293,37 +293,65 @@ def test_nat_converters(self):
...
+                                'nanos', 'n', 'NS', 'Nanoseconds',
+                                'Nanosecond', 'Nano', 'Nanos', 'N'], 'ns')])
+    def test_unit_parser(self, units, np_unit):
+        # validate all units, GH 6855, GH 21762
+        for unit in units:
+            # array-likes
+            expected = TimedeltaIndex([np.timedelta64(i, np_unit)
                                        for i in np.arange(5).tolist()])
-            tm.assert_index_equal(result, expected)
+            for wrapper in [np.array, list, pd.Index]:
```
Comment (jreback) [inline pandas/tests/scalar/timedelta/test_timedelta.py]: can you make this a parameter (call it box)

**Review by jreback — CHANGES_REQUESTED** [review]
test comment, otherwise lgtm.

File: pandas/tests/scalar/timedelta/test_timedelta.py (hunk 2 above)
Comment (sinhrks) [reply pandas/tests/scalar/timedelta/test_timedelta.py]: Thx, fixed.

**Review by jreback — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Personally, I think we should make a distinction between which units are accepted for the `unit` parameter, and which are parsed inside a string. 

For parsing a string, it is fine to accept both '15m', '15min', '15minute', '15minutes and '15T'. But for the `unit` keyword, is it needed to allow this flexibility? 
I would say that just the standard units (the numpy ones, as we had now) + maybe the abbreviations as used in our freq strings should be enough no? (so 'm' and 'T')

(and to be clear, the rest of this PR, fixing the inconsistency of what strings are parsed is of course totally fine!)

**Pull request merged**